src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaTypeModel.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public abstract class JavaTypeModel : IJavaResolvable
```

Comment:
Should this type also have a `JniFullName` property?  For class/interface types, this would be the normal `L…;` value, while for builtin types it could be the `I`/`C`/etc. value.  This would allow taking a `JavaTypeModel[]` for method parameters and constructing the JNI signature via `"(" + string.Join("", types => t.JniFullName) + ")" + returnType.JniFullName`.